peers/app_request_network.go

```diff
@@ -103,83 +100,99 @@ func NewNetwork(
 		return nil, nil, err
 	}
 
-	// We need to initially connect to some nodes in the network before peer
-	// gossip will enable connecting to all the remaining nodes in the network.
-	var beaconIPs, beaconIDs []string
-
-	peers, err := infoClient.Peers(context.Background())
-	if err != nil {
-		logger.Error(
-			"Failed to get peers",
-			zap.Error(err),
-		)
-		return nil, nil, err
+	arNetwork := &AppRequestNetwork{
+		Network:    testNetwork,
+		Handler:    handler,
+		infoClient: infoClient,
+		logger:     logger,
+		lock:       new(sync.Mutex),
 	}
 
-	// Randomly select peers to connect to until we have numInitialTestPeers
-	indices := rand.Perm(len(peers))
-	for _, index := range indices {
-		// Do not attempt to connect to private peers
-		if len(peers[index].PublicIP) == 0 {
-			continue
-		}
-		beaconIPs = append(beaconIPs, peers[index].PublicIP)
-		beaconIDs = append(beaconIDs, peers[index].ID.String())
-		if len(beaconIDs) == numInitialTestPeers {
-			break
+	// Manually connect to the validators of each of the source subnets.
+	// We return an error if we are unable to connect to sufficient stake on any of the subnets.
+	// Sufficient stake is determined by the Warp quora of the configured supported destinations,
+	// or if the subnet supports all destinations, by the quora of all configured destinations.
+	for _, sourceBlockchain := range cfg.SourceBlockchains {
```

Comment:
This loop body is pretty large. May want to consider moving parts out to helper functions.

Reply:
Would also be great to add unit tests for this since the logic is relatively complex. I'm not sure how difficult it is to mock the various clients and `arNetwork` though.